test/Microsoft.ComponentDetection.Detectors.Tests/Utilities/TestLogger.cs

```diff
@@ -0,0 +1,26 @@
+namespace Microsoft.ComponentDetection.Detectors.Tests.Utilities;
+
+using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+internal class TestLogger<T> : ILogger<T>, IDisposable
```

Comment:
Could you help me understand the need for this code? We have some tests that check the log output, and we have a utility that handles that functionality, which can be found at https://github.com/microsoft/component-detection/blob/dbcb973d640bc0529d58c646f94700df3f5147df/test/Microsoft.ComponentDetection.TestsUtilities/ExtensionMethods.cs#L27.